src/main/java/jenkins/plugins/logstash/configuration/ElasticSearch.java

```diff
@@ -0,0 +1,189 @@
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import hudson.Extension;
+import hudson.util.FormValidation;
+import hudson.util.Secret;
+import jenkins.plugins.logstash.Messages;
+import jenkins.plugins.logstash.persistence.ElasticSearchDao;
+
+public class ElasticSearch extends LogstashIndexer<ElasticSearchDao>
+{
+  private String username;
+  private Secret password;
+  private URL url;
+
+  @DataBoundConstructor
+  public ElasticSearch()
+  {
+  }
+
+  public URL getUrl()
+  {
+    return url;
+  }
+
+
+  private URI getUri()
+  {
+    if (url != null)
+    {
+      URI uri;
+      try
+      {
+        uri = url.toURI();
+        return uri;
+      }
+      catch (URISyntaxException e)
+      {
+        return null;
+      }
+    }
+    return null;
+  }
+
+  @DataBoundSetter
+  public void setUrl(URL uri)
+  {
+    this.url = uri;
+  }
+
+  public String getUsername()
+  {
+    return username;
+  }
+
+  @DataBoundSetter
+  public void setUsername(String username)
+  {
+    this.username = username;
+  }
+
+  public String getPassword()
+  {
+    return Secret.toString(password);
+  }
+
+  @DataBoundSetter
+  public void setPassword(String password)
+  {
+    this.password = Secret.fromString(password);
+  }
+
+  @Override
+  public boolean equals(Object obj)
+  {
+    if (obj == null)
+      return false;
+    if (this == obj)
+      return true;
+    if (getClass() != obj.getClass())
+      return false;
+    ElasticSearch other = (ElasticSearch) obj;
+    if (!Secret.toString(password).equals(other.getPassword()))
+    {
+      return false;
+    }
+    if (url == null)
+    {
+      if (other.url != null)
+        return false;
+    }
+    // String comparison is not optimal but comparing the urls directly  is
+    // criticized by findbugs as being a blocking operation
+    else if (!url.toString().equals(other.url.toString()))
+    {
+      return false;
+    }
+    if (username == null)
+    {
+      if (other.username != null)
+        return false;
+    }
+    else if (!username.equals(other.username))
+    {
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  public int hashCode()
+  {
+    final int prime = 31;
+    int result = super.hashCode();
+    result = prime * result + ((url == null) ? 0 : url.toString().hashCode());
+    result = prime * result + ((username == null) ? 0 : username.hashCode());
+    result = prime * result + Secret.toString(password).hashCode();
+    return result;
+  }
+
+  @Override
+  public ElasticSearchDao createIndexerInstance()
+  {
+    return new ElasticSearchDao(getUri(), username, Secret.toString(password));
+  }
+
+  @Extension
+  public static class ElasticSearchDescriptor extends LogstashIndexerDescriptor
+  {
+    @Override
+    public String getDisplayName()
+    {
+      return "Elastic Search";
+    }
+
+    @Override
+    public int getDefaultPort()
+    {
+      return 9300;
```

Comment:
umm.. isn't ES default port 9200?

Reply:
yes but with URI, we don't need this at all.